pkgs/tools/text/ripgrep/default.nix

```diff
@@ -28,10 +30,10 @@ rustPlatform.buildRustPackage rec {
 
   buildFeatures = lib.optional withPCRE2 "pcre2";
 
-  preFixup = ''
-    $out/bin/rg --generate man > rg.1
+  preFixup = lib.optionalString enableManpages ''
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/rg --generate man > rg.1
     installManPage rg.1
-
+  '' + ''
     installShellCompletion --cmd rg \
       --bash <($out/bin/rg --generate complete-bash) \
```

Comment:
Shouldn't an emulator be used here as well?

Right now this still gives me `/nix/store/xsy68p2vfcgflar0myiaz1d3na3rahym-stdenv-linux/setup: line 116: /nix/store/w09632mxy4jy9fiv4yn93h0a1saijh8v-ripgrep-aarch64-unknown-linux-gnu-14.0.1/bin/rg: cannot execute binary file: Exec format error` while building. It only doesn't fail because it's in a substitution.

Reply:
Argh, I just merged it. Let me know if there's an erratum to do, I'll take care of it immediately.

Reply:
thanks for catching this. followup PR is here: https://github.com/NixOS/nixpkgs/pull/271072